### PR TITLE
Update p2p proto definitons

### PR DESCRIPTION
**Pull request by omerfirmak**
and make each request use seperate pids

### CODE REVIEW

File: p2p/starknet/handlers.go [hunk 1]
```diff
@@ -43,133 +39,130 @@ func getBuffer() *bytes.Buffer {
...
 	return func() (proto.Message, bool) {
 		if count > 3 {
 			return nil, false
 		}
 		count++
-		return &spec.BlockHeader{
-			State: &spec.Merkle{
-				NLeaves: count - 1,
-			},
+		fmt.Println("counting ", count-1)
```
Comment (joshklop) [inline p2p/starknet/handlers.go:107]: Left over from debugging?

File: p2p/starknet/handlers.go [hunk 2]
```diff
@@ -43,133 +39,130 @@ func getBuffer() *bytes.Buffer {
...
-func (h *Handler) HandleGetBlocks(req *spec.GetBlocks) (Stream[proto.Message], error) {
+func (h *Handler) ReceiptsHandler(stream network.Stream) {
+	streamHandler[*spec.ReceiptsRequest](stream, h.OnReceiptsRequest, h.log)
+}
+
+func (h *Handler) TransactionsHandler(stream network.Stream) {
+	streamHandler[*spec.TransactionsRequest](stream, h.OnTransactionsRequest, h.log)
+}
+
+func (h *Handler) OnBlockHeadersRequest(req *spec.BlockHeadersRequest) (Stream[proto.Message], error) {
```
Comment (joshklop) [inline p2p/starknet/handlers.go:99]: Can we un-export the `On...Request` methods?

**Review by joshklop — APPROVED** [review]

File: p2p/starknet/handlers.go (hunk 1 above)
Comment (omerfirmak) [reply p2p/starknet/handlers.go]: yeap, removed.

File: p2p/starknet/handlers.go (hunk 2 above)
Comment (omerfirmak) [reply p2p/starknet/handlers.go]: done, might need to export them again at some point for testing

**Pull request merged**